src/settings/target/mod.rs

```diff
@@ -270,88 +251,73 @@ impl Manifest {
-        release: bool,
-    ) -> Result<(Option<String>, bool), failure::Error> {
-        let use_dot_dev_failure =
-            "Please specify the workers_dev boolean in the top level of your wrangler.toml.";
-        let use_dot_dev_warning =
-            format!("{}\n{} If you do not add workers_dev, this command may act unexpectedly in v1.5.0. Please see https://github.com/cloudflare/wrangler/blob/master/docs/content/environments.md for more information.", use_dot_dev_failure, emoji::WARN);
-        let wdd_failure = format!(
-            "{} Your environment should only include `workers_dev` or `route`. If you are trying to publish to workers.dev, remove `route` from your wrangler.toml, if you are trying to publish to your own domain, remove `workers_dev`.",
-            emoji::WARN
-        );
-
-        // TODO: deprecate --release, remove warnings and parsing
-        // switch wrangler publish behavior to act the same at top level
-        // and environments
-        // brace yourself, this is hairy
-        let workers_dev = match environment {
-            // top level configuration
-            None => {
-                if release {
-                    match self.workers_dev {
-                        Some(_) => {
-                            failure::bail!(format!("{} {}", emoji::WARN, use_dot_dev_failure))
-                        }
-                        None => {
-                            message::warn(&use_dot_dev_warning);
-                            false // wrangler publish --release w/o workers_dev is zoned deploy
-                        }
-                    }
-                } else if let Some(wdd) = self.workers_dev {
-                    if wdd {
-                        if let Some(route) = &self.route {
-                            if !route.is_empty() {
-                                failure::bail!(wdd_failure)
-                            }
-                        }
-                    }
-                    wdd
-                } else {
-                    message::warn(&use_dot_dev_warning);
-                    true // wrangler publish w/o workers_dev is zoneless deploy
-                }
-            }
+    ) -> Result<Option<String>, failure::Error> {
+        let conflicting_targets_failure = "Your environment should only include `workers_dev` or `route`. If you are trying to publish to workers.dev, remove `route` from your wrangler.toml, if you are trying to publish to your own domain, remove `workers_dev`.";
+        let pick_target_failure =
+            "You must specify either `workers_dev` or `route` and `zone_id` in order to publish.";
+        let empty_route_failure =
+            "If you want to deploy to workers.dev, remove `route` from your environment config.";
+
+        log::debug!("top level workers_dev: {:?}", self.workers_dev);
+        log::debug!("top level route: {:?}", self.route);
+
+        // start with top level configuration
```

Comment:
I still would LOVE a comment that displays all the expected combinations and outputs for the truth table... think you can slide that in before I approve this?

Reply:
oh oops - totally!